### PR TITLE
Add spec for Identity service bulk lookup

### PR DESCRIPTION
**Pull request by maxidor**

### CODE REVIEW

**Comment by richvdh** [comment]
@matrixbot: test this please

**Comment by richvdh** [comment]
@maxidor: this appears to be a superset of #1039: is that deliberate?

**Comment by maxidor** [comment]
@richvdh There is a shared appendices, but I didn't expect the PR to show up like this. Do you want me to split? This is still a logical piece if we close #1039

**Comment by richvdh** [comment]
No worries. if we handle #1039 then we can probably get this one into a state where it can be reviewed separately

**Comment by richvdh** [comment]
@dbkr perhaps you could look at this one when you're done with #1039 :innocent:

File: api/identity/lookup.yaml [hunk 1]
```diff
@@ -84,3 +84,56 @@ paths:
               signatures:
                 type: object
                 description: The signatures of the verifying identity services which show that the association should be trusted, if you trust the verifying identity services.
+  "/bulk_lookup":
+    post:
+      summary: Lookup Matrix user IDs for a list of 3pid.
```
Comment (dbkr) [inline api/identity/lookup.yaml]: 3pids (plural)

File: api/identity/lookup.yaml [hunk 2]
```diff
@@ -84,3 +84,56 @@ paths:
...
+            properties:
+              threepids:
+                type: array
+                items:
+                  type: array
+                  title: 3PID mappings
+                  items:
+                    type: string
+                    title: 3PID medium or address
+                description: an array of array containing the `3PID Medium type`_ in first position and the 3PID Address in second position.
```
Comment (dbkr) [inline api/identity/lookup.yaml]: array of arrays

**Comment by dbkr** [comment]
Oh, and also changelog entries would be great (we should get into the habit of adding them as we go)

**Comment by maxidor** [comment]
Signed-off-by: Maxime Dor @ Kamax.io (no email to avoid spam)

**Comment by maxidor** [comment]
@dbkr There is currently no changelog since this never had a released version. Not sure how to proceed further on that one.

**Comment by maxidor** [comment]
@dbkr Fixed spelling

**Comment by maxidor** [comment]
Closing following required #1039 being closed.

**Pull request closed without merging**